File: Code/app/src/main/java/net/listerily/moddedpepro/ManageNModsActivity.java
package net.listerily.moddedpepro;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.os.Build;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import net.listerily.moddedpepro.ui.BitmapRepeater;

public class ManageNModsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_manage_nmods);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        /*
        Bitmap bitmap = BitmapFactory.decodeResource(getResources(), R.drawable.activity_background);
        bitmap = BitmapRepeater.repeat(getWindowManager().getDefaultDisplay().getWidth(), getWindowManager().getDefaultDisplay().getHeight(), bitmap);
        getWindow().getDecorView().setBackground(new BitmapDrawable(bitmap));
        */
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        if(item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: ModdedBE-Android/app/src/main/java/net/listerily/moddedbe/ManageNModsActivity.kt
package net.listerily.moddedbe

import android.content.Context
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.CompoundButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SwitchCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.endercore.android.EnderCore
import org.endercore.android.nmod.NMod
import org.endercore.android.operator.NModManager
import java.io.FileNotFoundException

class ManageNModsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_manage_nmods)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        val recyclerView = findViewById<RecyclerView>(R.id.manageNModsRecycler)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = NModsAdapter(EnderCore.instance.nModManager, this)
    }

    private class NModViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var rootView = itemView
    }

    private class NModsAdapter(mgr: NModManager, ctx: Context) : RecyclerView.Adapter<NModViewHolder>() {
        private val allNMods: ArrayList<NMod> = mgr.allNMods
        private val context = ctx
        private val manager = mgr
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NModViewHolder {
            val viewItem = LayoutInflater.from(context).inflate(R.layout.recycler_nmod_item, parent, false)
            return NModViewHolder(viewItem)
        }

        override fun getItemCount(): Int {
            return allNMods.size
        }

        override fun onBindViewHolder(holder: NModViewHolder, position: Int) {
            val view = holder.rootView
            val nmod = allNMods[position]

            val textViewName = view.findViewById<TextView>(R.id.textViewNModName)
            textViewName.text = nmod.name

            val imageViewIcon = view.findViewById<ImageView>(R.id.imageViewNModIcon)
            var icon: Drawable? = null
            if (nmod.packageManifest.icon != null)
                try {
                    icon = Drawable.createFromStream(nmod.openInFiles(nmod.packageManifest.icon), nmod.packageManifest.icon)
                } catch (ignored: FileNotFoundException) {
                }
            if (icon == null)
                imageViewIcon.visibility = View.INVISIBLE
            else {
                imageViewIcon.visibility = View.VISIBLE
                imageViewIcon.setImageDrawable(icon)
            }

            val buttonRemove = view.findViewById<Button>(R.id.buttonRemoveNMod)
            buttonRemove.setOnClickListener {
                run {
                    manager.uninstallNMod(nmod.uuid)
                    notifyItemRemoved(position)
                }
            }

            val switchEnabled = view.findViewById<SwitchCompat>(R.id.switchNModEnabled)
            switchEnabled.isChecked = manager.isNModEnabled(nmod)
            switchEnabled.setOnCheckedChangeListener { _: CompoundButton, checked: Boolean ->
                run {
                    manager.setNModEnabled(nmod, checked)
                }
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }
}